forest/benchmarking/direct_fidelity_estimation.py

```diff
@@ -283,44 +283,44 @@ def monte_carlo_process_dfe(program: Program, qubits: List[int], benchmarker: Be
         program=program, qubits=qubits)
-    return DFEexperiment(exp,'process')
+    return DFEExperiment(exp, 'monte carlo, process')
 
 
-def acquire_dfe_data(qc: QuantumComputer, dfe: DFEexperiment, n_shots = 10_000, active_reset=False):
+def acquire_dfe_data(qc: QuantumComputer, dfe: DFEExperiment, n_shots = 10_000, active_reset=False):
     # TODO: add number of shots
     #       add variance bounds?
-    res = list(measure_observables(qc, dfe.exp, n_shots=n_shots, active_reset=active_reset))
-    return DFEdata(res, 
-                   in_states = [str(exp[0].in_state) for exp in dfe.exp],
-                   program = dfe.exp.program,
-                   out_pauli = [str(exp[0].out_operator) for exp in dfe.exp], 
-                   pauli_point_est = np.array([r.expectation for r in res ]), 
-                   pauli_var_est = np.array([r.stddev**2 for r in res]), 
-                   cal_point_est = np.array([r.calibration_expectation for r in res ]), 
-                   cal_var_est = np.array([r.calibration_stddev**2 for r in res ]), 
-                   dimension = 2**len(dfe.exp.qubits), 
-                   qubits = dfe.exp.qubits,
-                   type = dfe.type)
-
-
-def analyse_dfe_data(data: DFEdata):
-    mean = np.sum([r.expectation for r in data.res])
-    variance = np.sum([r.stddev**2 for r in data.res])
+    res = list(measure_observables(qc, dfe.experiments, n_shots=n_shots, active_reset=active_reset))
+    return DFEData(results = res,
+                   in_states = [str(exp[0].in_state) for exp in dfe.experiments],
+                   program = dfe.experiments.program,
+                   out_pauli = [str(exp[0].out_operator) for exp in dfe.experiments],
+                   pauli_point_est = np.array([r.expectation for r in res ]),
+                   pauli_var_est = np.array([r.stddev**2 for r in res]),
+                   cal_point_est = np.array([r.calibration_expectation for r in res ]),
+                   cal_var_est = np.array([r.calibration_stddev**2 for r in res ]),
+                   dimension = 2**len(dfe.experiments.qubits),
+                   qubits = dfe.experiments.qubits,
+                   kind = dfe.kind)
+
+
+def analyse_dfe_data(data: DFEData) -> DFEEstimate:
+    p_mean = np.mean(pauli_point_est)
+    p_variance = np.sum(pauli_var_est)
     d = data.dimension
 
-    if data.type == 'state':
-        mean_est = np.mean([r.expectation for r in data.res])
-        var_est = np.sum([r.stddev**2 for r in data.res])/len(data.res)**2
-    elif data.type == 'process':
-        mean_est = (d**2 * np.mean([r.expectation for r in data.res]) + d)/(d**2+d)
-        v = np.sum([r.stddev**2 for r in data.res])/len(data.res)**2
+    if data.kind == 'state':
+        mean_est = p_mean
+        var_est = p_variance / len(data.results) ** 2
+    elif data.kind == 'process':
+        mean_est = (d**2 * p_mean + d)/(d**2+d)
+        v = p_variance / len(data.results) ** 2
```

Comment:
`if data.kind.split(', ')[-1] == 'state'`    ( or 'process')

Also this is minor but I find this calculation a bit confusing, I think mostly because p_mean and p_variance are initially 'normalized' differently. Would it make sense to do:

```
mean_est = np.mean(pauli_point_est)
p_variance = np.sum(pauli_var_est)/len(data.results) ** 2

d = data.dimension
if data.kind.split(', ')[-1] == 'process':
    mean_est = mean_est * d**2 / (d**2 + d) + 1/(d+1)
    var_est *= d**2/(d+1)**2
```

Reply:
I pushed new changes that address this.